src/main/java/com/gmail/goosius/siegewar/utils/PermsCleanupUtil.java

```diff
@@ -0,0 +1,114 @@
+package com.gmail.goosius.siegewar.utils;
+
+import com.gmail.goosius.siegewar.SiegeWar;
+import com.palmergames.bukkit.config.CommentedConfiguration;
+import com.palmergames.bukkit.towny.permissions.TownyPerms;
+
+import java.util.List;
+
+public class PermsCleanupUtil {
+
+    public static void cleanupPerms(boolean siegeWarPluginError) {
+        if(siegeWarPluginError) {
+            SiegeWar.severe("SiegeWar is in safe mode. Perms cleanup not attempted.");
+            return;
+        }
+        CommentedConfiguration townyPermsFile = TownyPerms.getTownyPermsFile();
+        boolean success = false;
+        success = deleteMilitaryPermsFromSheriffRank(townyPermsFile, success);
+        success = deleteMilitaryPermsFromAssistantRanks(townyPermsFile, success);
+        success = deleteFireCannonPermFromGuardRank(townyPermsFile, success);
+        success = deleteGunnerAndEngineerRanks(townyPermsFile, success);
```

Comment:
I should have had this included earlier but this should probably be something like success = deleteblah || deleteblah || deleteblah; instead of being separate, otherwise a true could be overridden by a later false.